### ISSUE TITLE
[20230409] Weekly AI ArXiv 만담 시즌2 - 13회차

### ISSUE THREAD

**Issue by veritas9872** [jungwoo-ha/WeeklyArxivTalk#79]
**뉴스**:
“Gemini”: Google and Deepmind develop GPT-4 competition
URL: https://the-decoder.com/gemini-google-and-deepmind-develop-gpt-4-competition

Twitter:
![image](https://user-images.githubusercontent.com/33523965/230725186-67d6fffd-60dd-4738-861d-cf13768bf7a7.png)

구글과 딥마인드에서 GPT-4에 대항하기 위해 multi-trillion parameter model을 학습하는 중이라는 소문이 업계에서 빈번하게 들려옵니다. 심지어 Jeff Dean 옹께서 직접 코딩을 하신다고 하는데 그만큼 구글에서 LLM 부분에서 뒤쳐지지 않기 위해 올인하는 것을 볼 수 있습니다.

**연구**:

**Segment Anything**

Blog: https://segment-anything.com
GitHub: https://github.com/facebookresearch/segment-anything
ArXiv: https://arxiv.org/abs/2304.02643

![image](https://user-images.githubusercontent.com/33523965/230726387-851bf735-c963-476c-84e6-5ec69b45aeae.png)

Meta에서 image segmentation을 위한 거대 모델을 코드와 model weight를 포함해서 오픈소스했습니다.

학습을 위해서 모델 스스로 라벨을 생성해서 학습에 포함하는 방법을 통해 처음에는 human segmentation의 도움을 받았지만 학습이 진행할수록 human labelling을 축소하는 방식을 적용하는 데이터 엔진을 구성했습니다. 단순히 클릭하는 것과 언어 프롬프트를 통해 물체의 영역을 분할할 수 있어 여러 분야에 범용적으로 적용될 것으로 생각됩니다.

**Self-Refine: Iterative Refinement with Self-Feedback**

Blog: https://selfrefine.info
GitHub: https://github.com/madaan/self-refine
ArXiv: https://arxiv.org/abs/2303.17651

![image](https://user-images.githubusercontent.com/33523965/230726596-f96297cf-d16c-4b34-a4d4-ba3f7df27eb1.png)

ChatGPT 및 대화형 거대 언어 모델이 지금까지 Reinforcement Learning with Human Feedback (RLHF)를 사용해왔지만 고품질의 human labelling이 필요하다는 단점이 있습니다. 이미 학습이 완료된 다른 모델에서 라벨을 생성하는 방법도 가능하지만 초거대 모델을 처음 개발하는 입장에서는 다른 방법이 있으면 도입이 더 쉬워질 것으로 생각됩니다.

해당 논문에서는 pre-trained model에 직접 RLHF를 진행하는 대신 refinement와 feedback prompt를 제시하는 방식을 통해 답변을 점진적으로 고도화합니다. 이것은 인간이 사고하는 방식과 유사하고 한 번 실수한 것을 수정할 기회를 준다는 장점이 있지만 최종 결과를 위해 즉답을 할 수 없어 real-time으로 답변하기 어려워진다는 단점이 있습니다.

Technical Papers:

**TPU v4: An Optically Reconfigurable Supercomputer for Machine Learning with Hardware Support for Embeddings**
Blog: https://cloud.google.com/blog/topics/systems/tpu-v4-enables-performance-energy-and-co2e-efficiency-gains?hl=en
ArXiv: https://arxiv.org/abs/2304.01433

![image](https://user-images.githubusercontent.com/33523965/230725399-fdd7d8af-f2ff-478c-bf00-69791f46c5d3.png)

구글에서 딥러닝에 특화된 TPU v4의 구조를 설명하는 white paper를 공개했습니다. 기존에 비해 가장 유의할 부분은 데이터 전송에서 광학을 이용함으로써 전력 소모, 시간, 네트워크 구성 편의성 등 여러 장점을 살렸다는 것입니다. ~그러나 가격은 Infiniband의 10배라고 합니다. 역시 돈이 좋습니다...~

전부터 광학을 사용하는 것이 데이터 전송 속도 병목의 좋은 해결책이라고 생각을 했었는데 연산 회로에서는 아니지만 이미 데이터 전송에서 사용되고 있는 것을 보아 추후 더 중요해질 것으로 생각됩니다.

**BloombergGPT: A Large Language Model for Finance**
Blog: https://www.bloomberg.com/company/press/bloomberggpt-50-billion-parameter-llm-tuned-finance
ArXiv: https://arxiv.org/abs/2303.17564

<img width="441" alt="image" src="https://user-images.githubusercontent.com/33523965/230725713-7a9a0208-4799-4b1c-9c33-813660eee6eb.png">

금융 정보 분야의 대표주자인 Bloomberg에서 금융 분야에 특화된 거대 언어모델을 학습해 서비스에 제공할 예정입니다.
기존의 task-specific한 모델과 다르게 대량의 financial data를 전체 학습 데이터의 절반 이상 차지할 정도로 많이 사용하고 금융 데이터에서 사용되는 정보의 구성 등에서 많은 전문성을 요구했습니다. 또한, AWS에서 비교적 적은 리소스로 50B 모델을 학습할 수 있었는데 GPT API에 의존하지 않고도 데이터만 충분하다면 분야에 특화된 모델을 만들 수 있는 것을 보여주었습니다. 추후 대화형 인공지능 모델의 비즈니스 모델 개발에서 중요한 요소일 것으로 생각됩니다.

**Comment by nick-jhlee** [comment]
# 알쓸신뉴스..
- Launch of **Continual AI Unconference (CLAI Unconf)**!
  - https://continualai.github.io/
  - 24hr long virtual *conference*
  - proceeding to be available at PMLR
  - "preregistration" with idea and write the actual paper during conf?!
  - twitter link: https://twitter.com/continualai/status/1641092869728468993?s=46&t=llKohaNYR1IR_yaWlq40TA

**Comment by terryum** [comment]
### 다양한 GPT 어플리케이션들의 출시

[SkinChat.ai](https://skinchat.ai?utm_source=wtm&utm_medium=qr_code&utm_campaign=wtm_meetup)
: ART Lab에서 SkinGPT를 활용한 SkinChat.ai 를 선보였습니다 [[관련 페이스북 글]](https://www.facebook.com/terry.artlab/posts/pfbid02bixv9T2GpfwDdBdXHaLwAhvRhN6J5d6ojkjeRRJ7MNs3mPLPZqGJnfzyrssLknPcl)
- GPT + Vision + Domain Knowledge (Skin+Cosmetics)
- GPT의 약점은 1) 전문성 2) 신뢰성 3) 최신성 4) 사용성 5) 멀티모달 인풋이 부족하단 점인데, 이 부분을 개선하기 위해 노력했습니다 (3주만에 만든 MVP라 매달 업데이트 될 예정입니다)
- 인도에서의 반응이 좋습니다. (국내는 "뷰티"적 접근, 해외는 "헬스케어"적 접근)

-  전문지식+GPT, 검색+GPT 등은 강력한 비즈니스 툴을 만들어 줍니다.
- 앞으로는 태스크를 문서화하는 것 또는 대화 데이터를 쌓는 것이 곧 AI를 만드는 길이 될 것입니다. (e.g. [주님AI](https://askjesus.oopy.io/))
- 누구나 해킹할 수 있는 시대, 새로운 LLM 보안이 필요합니다.

**Comment by kimyoungdo0122** [comment]
### News
- [골드만삭스는 AI로 인해 일자리 3억 개가 줄어들 것으로 예측](https://www.forbes.com/sites/jackkelly/2023/03/31/goldman-sachs-predicts-300-million-jobs-will-be-lost-or-degraded-by-artificial-intelligence/?sh=7215cb0782b4&utm_campaign=Artificial%2BIntelligence%2BWeekly&utm_medium=web&utm_source=Artificial_Intelligence_Weekly_327)
    - [자료 링크](https://www.key4biz.it/wp-content/uploads/2023/03/Global-Economics-Analyst_-The-Potentially-Large-Effects-of-Artificial-Intelligence-on-Economic-Growth-Briggs_Kodnani.pdf)
    - 전통적으로, 자동화 기술은 일자리에 큰 변화를 가져왔음
    - 자동화로 인해 줄어든 일자리만큼 새로운 일자리가 창출되었으나, 
    AI로 인한 일자리 감소는 더 많은 일자리 창출로 이어질지 미지수
    - 그동안 기계의 발전이 단순 노동을 대체해왔다면, AI는 이제 모든 직무에 영향을 미칠 예정

- [게티이미지의 Stability AI에 대한 소송](https://arstechnica.com/tech-policy/2023/02/getty-sues-stability-ai-for-copying-12m-photos-and-imitating-famous-watermark/)
    - Stability AI의 Stable Diffusion이 게티 이미지의 소스들을 무단 스크래핑해서 모델 학습에 사용했다는 주장
    - 생성된 이미지가 양질일수록 높은 이미지 퀄리티를 유지해왔던 게티 이미지와 아티스트들의 비중이 많이 들어간다고(?)
    - 두 번째 소송인데, 판결이 나오기까진 오래 걸리지 않을까..

**Comment by snoop2head** [comment]
### Open Assistant
- Application: https://open-assistant.io/chat
- Conversation Corpus: https://huggingface.co/datasets/OpenAssistant/oasst1
- Code: https://github.com/LAION-AI/Open-Assistant
- Model Checkpoint: https://huggingface.co/OpenAssistant
- Paper? Technical Report?: https://www.ykilcher.com/OA_Paper_2023_04_15.pdf

LAION AI가 Pythia와 LLaMA 기반으로 10B+ 모델들을 제작하고 공개했습니다. 특히 LLaMA 모델을 만드는 모든 과정에 사용된 재료들을 공개한 것 같습니다. 리포트에 따르면 Pythia-12B만 해도 93.5% preferable as GPT-3.5.turbo라는데, 사용해보면서 확인할 예정입니다.

### 주요 훈련 기법
- Supervised Fine-tuning (SFT)
- Reinforcement learning from human feedback (RLHF)

특히 Reward Model인 Deberta checkpoint도 공개한 덕분에 이쪽 분야 공부하는데에 도움이 될 것 같네요. 

### 데이터셋
161,443개의 메세지와 461,292개의 메세지 퀄리티 평점으로 구성되어 있습니다. 주요 언어는 영어와 스페인어(영어 40%, 스페인어 30%)이며, 현재 어플리케이션에서 한국어로 인사를 해도 영어로 답변이 나옵니다. 

<img width="808" alt="image" src="https://user-images.githubusercontent.com/28736695/232258007-2e887a4a-a26d-4a98-bca4-07520258aeee.png">

Conversation Tree라는 자료구조로 데이터셋을 구성했는데, prompt-assistant가 대화를 주고받는 방식으로 구성이 되어 있습니다. 이때 각 Node는 반드시 인간일 필요는 없으며, 기계의 데이터도 어느 정도 들어간 것으로 보입니다. Root Node - Leaf Node로 향하는 각 path를 thread로 정의했습니다. 161,443개의 메세지는 약 92,000개의 prompt message와 약 69,600개의 assistant reply로 구성되어 있습니다
